apis/apps/v1alpha1/backupschedule_types.go

```diff
@@ -20,54 +20,88 @@ import (
+// +kubebuilder:resource:shortName="nsb"
+// +kubebuilder:printcolumn:name="Schedule",type=string,JSONPath=`.spec.schedule`,description="The current schedule set for the scheduled backup"
+// +kubebuilder:printcolumn:name="Pause",type=string,JSONPath=`.spec.pause`,description="Whether or not the scheduled backup is paused"
+// +kubebuilder:printcolumn:name="Last Triggered Backup",type=string,JSONPath=`.status.lastScheduledBackupTime`,description="The timestamp at which the last backup was triggered"
+// +kubebuilder:printcolumn:name="Last Successful Backup",format=date-time,type=string,JSONPath=`.status.lastSuccessfulBackupTime`,description="The timestamp at which the last backup was successful completed"
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
 
-type BackupSchedule struct {
+type NebulaScheduledBackup struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   BackupScheduleSpec   `json:"spec,omitempty"`
-	Status BackupScheduleStatus `json:"status,omitempty"`
+	Spec   ScheduledBackupSpec   `json:"spec,omitempty"`
+	Status ScheduledBackupStatus `json:"status,omitempty"`
 }
 
 // +kubebuilder:object:root=true
-// BackupScheduleList contains a list of BackupSchedule.
-type BackupScheduleList struct {
+// NebulaScheduledBackupList contains a list of NebulaScheduledBackup.
+type NebulaScheduledBackupList struct {
 	metav1.TypeMeta `json:",inline"`
 	metav1.ListMeta `json:"metadata,omitempty"`
 
-	Items []BackupSchedule `json:"items"`
+	Items []NebulaScheduledBackup `json:"items"`
 }
 
-// BackupScheduleSpec contains the specification for a backupSchedule of a nebula cluster backupSchedule.
-type BackupScheduleSpec struct {
+// ScheduledBackupSpec contains the specification for a NebulaScheduledBackup of a nebula cluster NebulaScheduledBackup.
+type ScheduledBackupSpec struct {
 	// Schedule specifies the cron string used for backup scheduling.
 	Schedule string `json:"schedule"`
 	// Pause means paused backupSchedule
-	Pause bool `json:"pause,omitempty"`
-	// MaxBackups is to specify how many backups we want to keep
-	// 0 is magic number to indicate un-limited backups.
-	// if MaxBackups and MaxReservedTime are set at the same time, MaxReservedTime is preferred
+	Pause *bool `json:"pause,omitempty"`
+	// MaxBackups specifies how many backups we want to keep in the remote storage bucket.
+	// 0 is the magic number to indicate unlimited backups.
+	// if both MaxBackups and MaxReservedTime are set at the same time, MaxReservedTime will be used
 	// and MaxBackups is ignored.
 	MaxBackups *int32 `json:"maxBackups,omitempty"`
-	// MaxReservedTime is to specify how long backups we want to keep.
-	MaxReservedTime *string `json:"maxReservedTime,omitempty"`
-	// BackupTemplate is the specification of the backup structure to get scheduled.
+	// MaxRetentionTime specifies how long we want the backups in the remote storage bucket to be kept for.
```

Comment:
Shouldn't it be the case that if either of these two parameters meets the condition, it should be deleted? Instead of setting both parameters and using only one as the basis for judgement.

Reply:
The current design is to use MaxReservedTime if both are specified as this is simpler understand for the customer and easier to implement. Otherwise we don't know whether the customer means "delete a backup if there are more then MaxBackups **and** the backup is older than MaxReservedTime" or "delete a backup if there are more then MaxBackups **or** if the backup if older than MaxReservedTime". 

We can definitely think about which version we want to support and change the design to accommodate both parameters in a future version if there's customer demand for this. What do you think @MegaByte875

Reply:
Here's an example for a simple task: a log cleaning function. There are two settings: one for the maximum number of days to be retained, and one for the maximum storage to be kept. If a user configures both, how should we proceed?

I assume that the user wants to keep a certain number of days, but they also don't want to consume too much storage.

Returning to this issue, I think it's similar. By the way, if the user only wants to retain a certain number of days, why would they configure MaxBackups?